test/test_torch.py

```diff
@@ -697,9 +697,33 @@ def _testSelection(self, torchfn, mathfn):
     def test_max(self):
         self._testSelection(torch.max, max)
 
+    @staticmethod
+    def _test_max_with_inf(self, conv_fn):
+        a = torch.tensor([[float('inf')] * 3] * 2)
```

Comment:
This doesn't test for the bug:

1) The tensor should be filled with -inf for max and +inf for min
2) You need to apply conv_fn to `a` so that a is a cuda tensor in the CUDA test.